pkg/discovery/agent.go

```diff
@@ -32,9 +34,8 @@ type Agent struct {
 }
 
 // NewAgent creates a new agent
-func NewAgent(cfg *config.AgentConfig, client anypoint.Client) (agent *Agent) {
+func NewAgent(cfg *config.AgentConfig, client anypoint.Client, sm *subscription.Manager) (agent *Agent) {
```

Comment:
subscription.Manager should implement an interface, and NewAgent should accept that interface for this parameter so that it doesn't rely on the type, making it easier to test.